client/js/components/fragments/MiniProfile.js

```diff
@@ -1,15 +1,35 @@
 import React, { Component, PropTypes } from 'react';
 import { Link } from 'react-router';
 
+import DefaultAvatar from '../../../img/avatar.jpg';
+import 'file?name=[name].[ext]!../../../img/avatar.jpg';
+
 class MiniProfile extends Component {
   render() {
     const { user } = this.props;
 
+    let fullName;
+    if (user.firstName && user.lastName) {
+      fullName = `${user.firstName} ${user.lastName} (${user.email})`;
+    } else if (!user.firstName && user.lastName) {
+      fullName = `${user.lastName} (${user.email})`;
+    } else if (user.firstName && !user.lastName) {
+      fullName = `${user.firstName} (${user.email})`;
+    } else {
+      fullName = user.email;
+    }
+
+    let avatar;
```

Comment:
Could you move that logic to separate function? So that render() is as simple as possible